dataplane/internal/engine/engine.go

```diff
@@ -353,11 +355,21 @@ func (e *Engine) addNextHopList(ctx context.Context, nhg *dpb.NextHopList) ([]*f
 }
 
 // addNextHopGroupIDList adds an entry to the next hop group table.
-func (e *Engine) addNextHopGroupIDList(ctx context.Context, id uint64, nhg *dpb.NextHopIDList) error {
+func (e *Engine) addNextHopGroupIDList(ctx context.Context, id uint64, nhg *dpb.NextHopIDList, mode dpb.GroupUpdateMode) error {
+	e.nextHopGroupsMu.Lock()
+	defer e.nextHopGroupsMu.Unlock()
+
+	hops := nhg.GetHops()
+	weights := nhg.GetWeights()
+	if mode == dpb.GroupUpdateMode_GROUP_UPDATE_MODE_APPEND {
+		hops = append(hops, e.nextHopGroups[id].Hops...)
+		weights = append(weights, e.nextHopGroups[id].Weights...)
```

Comment:
nit: this seems to be prepending instead of appending the new hops which is initially confusing.

Reply:
changed